specs/consensus.md

```diff
@@ -71,16 +85,149 @@ Consensus Rules
+
+Evidence is the first set of state transitions that are applied, and represent proof of validator misbehavior.
+
+### `block.availableData.transactionData`
+
+Once [evidence has been processed](#blockavailabledataevidencedata), transactions are applied to the state. Note that _transactions_ mutate the state (essentially, the validator set and minimal balances), while _messages_ do not. See [the architecture documentation](./architecture.md) for more info.
+
+`block.availableData.transactionData` is simply a list of [WrappedTransaction](./data_structures.md#wrappedtransaction)s. For each wrapped transaction in this list, `wrappedTransaction`, with index `i` (starting from `0`), the following checks must be `true`:
+
+1. `wrappedTransaction.index` == `i`.
+
+For `wrappedTransaction`'s [transaction](./data_structures.md#transaction) `transaction`, the following checks must be `true`:
+
+1. `transaction.signature` must be a [valid signature](./data_structures.md#public-key-cryptography) over `transaction.signedTransactionData`.
+
+Finally, each `wrappedTransaction` is processed depending on [its transaction type](./data_structures.md#signedtransactiondata). These are specified in the next subsections, where `tx` is short for `transaction.signedTransactionData`, and `sender` is the recovered signing [address](./data_structures.md#address). After applying a transaction, the new state state root is computed.
+
+#### SignedTransactionDataTransfer
+
+The following checks must be `true`:
+
+1. `tx.type` == [`TransactionType.Transfer`](./data_structures.md#signedtransactiondata).
+1. `tx.amount` <= `state.accounts[sender].balance`.
+1. `tx.nonce` == `state.accounts[sender].nonce + 1`.
+
+Apply the following to the state:
+
+```
+state.accounts[sender].nonce += 1
+
+state.accounts[sender].balance -= tx.amount
+state.accounts[tx.to].balance += tx.amount
 ```
+
+#### SignedTransactionDataPayForMessage
+
+The following checks must be `true`:
+
+1. `tx.type` == [`TransactionType.PayForMessage`](./data_structures.md#signedtransactiondata).
+1. `tx.nonce` == `state.accounts[sender].nonce + 1`.
+1. `uint(tx.messageNamespaceID) % 2` == `0`. In other words, the namespace ID interpreted as an unsigned integer is even.
+1. The `ceil(tx.messageSize / SHARE_SIZE)` shares starting at index `wrappedTransactions.messageStartIndex` must:
+    1. Have namespace ID `tx.messageNamespaceID`.
+1. `tx.messageShareCommitment` == computed as described [here](./data_structures.md#signedtransactiondatapayformessage).
+
+Apply the following to the state:
+
+```
+state.accounts[sender].nonce += 1
+```
+
+#### SignedTransactionDataCreateValidator
+
+The following checks must be `true`:
+
+1. `tx.type` == [`TransactionType.CreateValidator`](./data_structures.md#signedtransactiondata).
+1. `tx.amount` <= `state.accounts[sender].balance`.
+1. `tx.nonce` == `state.accounts[sender].nonce + 1`.
+1. `tx.commissionRate` <!--TODO check some bounds here-->
```

Comment:
Add todo as issues

Reply:
I think John already opened issues for them, e.g.: https://github.com/lazyledger/lazyledger-specs/issues/95
These could be a bit more descriptive IMO but I guess that's OK.

Reply:
Yeah, the rest of the TODOs were removed in 1e2df94, except for this one so that I don't forget since it's in the middle of a bunch of other text as opposed to a discrete concept.